### PR TITLE
NMS-9293: JDBCDataCollector : do not change custom resourcetype when query returns only one row

### PR DESCRIPTION
**Pull request by jm-k**

### CODE REVIEW

File: features/jdbc-collector/src/main/java/org/opennms/netmgt/collectd/JdbcCollector.java [hunk 1]
```diff
@@ -241,8 +241,11 @@ public CollectionSet collect(CollectionAgent agent, EventProxy eproxy, Map<Strin
                             JdbcCollectionResource resource = null;
                             
                             // Create the appropriate resource container.
-                            if(singleInstance) {
-                                resource = nodeResource;
+                            if(null==query.getResourceType()) {
```
Comment (j-white) [inline features/jdbc-collector/src/main/java/org/opennms/netmgt/collectd/JdbcCollector.java]: By convention, we typically allow a resource type of `"node"` to refer to node level resources i.e. https://github.com/OpenNMS/opennms/blob/develop/integrations/opennms-vmware/src/main/java/org/opennms/netmgt/collectd/VmwareCollector.java#L247

We should also consider this in addition to checking for `null`.

File: features/jdbc-collector/src/main/java/org/opennms/netmgt/collectd/JdbcCollector.java [hunk 2]
```diff
@@ -241,8 +241,11 @@ public CollectionSet collect(CollectionAgent agent, EventProxy eproxy, Map<Strin
...
+                                if (!singleInstance){
+                                    LOG.warn("Multiple row results but there is no resourceType defined : only first row will be persisted");
+                                    resource = nodeResource;
+                                }
```
Comment (j-white) [inline features/jdbc-collector/src/main/java/org/opennms/netmgt/collectd/JdbcCollector.java]: Won't resource be `null` if resource type is defined, and this is a single instance?

**Comment by j-white** [comment]
Closing due to lack of activity. PR is still pending changes.

**Comment by jb4415** [comment]
How do we get this change pushed into production?

**Pull request closed without merging**